.github/workflows/release.yml

```diff
@@ -0,0 +1,32 @@
+name: Create a draft release
+
+on:
+  push:
+    branches: [main]
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  autorelease:
+    name: Create release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Release Notes
+        run: |
+          git log $(git describe HEAD~ --tags --abbrev=0)..HEAD --pretty='format:* %h %s%n' --no-merges >> ".github/release_template.md"
```

Comment:
again, we have a changelog we maintain manually at this stage, and which contains a cleaned-up version of the changes (this is goin got get a lot of noise), why not use that instead?

Reply:
Updated the workflow to use an action that extracts release notes from changelog when creating a release
https://github.com/marketplace/actions/extract-release-notes

Reply:
oooh this is even better better than the script! thanks for sharing this! Resolving